titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -259,6 +278,36 @@ public void launchTasks(List<TaskInfoRequest> requests, List<VirtualMachineLease
         }
     }
 
+    private void launchTasksAsync(List<TaskInfoRequest> requests) {
+        List<Mono<Void>> podAddActions = new ArrayList<>(requests.size());
+        for (TaskInfoRequest request : requests) {
+            V1Pod v1Pod = taskInfoToPod(request);
+            Mono<Void> podAddAction = KubeUtil
+                    .<V1Pod>toReact(handler -> kubeApiFacade.getCoreV1Api().createNamespacedPodAsync(
+                            KUBERNETES_NAMESPACE, v1Pod, null, null, null, handler
+                    ))
+                    .doOnSubscribe(subscription -> {
+                        launchTaskCounter.increment();
+                        logger.info("creating pod: {}", v1Pod);
+                        podSizeMetrics.record(KubeUtil.estimatePodSize(v1Pod));
+                    })
+                    .timeout(Duration.ofMillis(directKubeConfiguration.getKubeApiClientTimeoutMs()))
+                    .ignoreElement()
+                    .cast(Void.class)
+                    .onErrorResume(error -> {
+                        logger.error("Unable to create pod with error: {}", KubeUtil.toErrorDetails(error), error);
+                        return Mono.empty();
+                    });
+            podAddActions.add(podAddAction);
+        }
+
+        try {
+            Flux.merge(podAddActions).blockLast();
```

Comment:
consider adding a concurrency limit to the `merge` so we avoid overloading kube-apiserver during big bursts

Reply:
also consider `mergeDelayError` to allow partial failures mirroring the previous behavior